packages/components/src/components/primitives/Tag/Tag.tsx

```diff
@@ -4,40 +4,24 @@ import {
   TagLeftIcon,
   TagRightIcon,
   TagLabel,
-  useStyleConfig,
+  useStyles,
+  forwardRef,
+  StyleProps,
 } from '@chakra-ui/react';
-import { TagProps, TagStyle } from './types';
+import { TagProps, TagStaticMembers } from './types';
 
-export const Tag: FC<TagProps> = ({
-  label,
-  icon,
-  iconPosition,
-  size,
-  ...rest
-}) => {
-  const {
-    container,
-    label: labelStyle,
-    leftIcon,
-    rightIcon,
-  } = useStyleConfig('Tag', { size }) as TagStyle;
-  return (
-    <ChakraTag {...container} {...rest}>
-      {icon && iconPosition !== 'right' && (
-        <TagLeftIcon
-          data-testid="cmpsr.tag.left-icon"
-          {...leftIcon}
-          as={icon}
-        />
-      )}
-      <TagLabel {...labelStyle}>{label}</TagLabel>
-      {icon && iconPosition === 'right' && (
-        <TagRightIcon
-          data-testid="cmpsr.tag.right-icon"
-          {...rightIcon}
-          as={icon}
-        />
-      )}
-    </ChakraTag>
-  );
-};
+export const Tag: FC<TagProps> & TagStaticMembers = (props) => <ChakraTag {...props} />;
```

Comment:
this has to be a forwardRef

Reply:
good catch, I had forgotten it, sorry, done!

Reply:
About the storybook file, I hadn't realized that I had included it in the commit, my bad, just keep in mind that this file won't work since it is using the old version of the tag component.

Reply:
you have to apply the fixes to the storybook to work, that's evident, but do not modify the existing stories other adapting the code just to verify that the refactor doesn't affect the behaviour of the component

Reply:
done